src/app/features/sign-up/sign-up.component.ts

```diff
@@ -1,30 +1,60 @@
 import { Component } from '@angular/core';
-import { FormControl } from '@angular/forms';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
 import { Router } from '@angular/router';
 import { AuthService } from 'src/app/core/auth.service';
+import { matchValidator } from 'src/app/core/matchValidator.directive';
+import { passwordValidator } from 'src/app/core/passwordValidator.directive';
 
 @Component({
   selector: 'app-sign-up',
   templateUrl: './sign-up.component.html',
   styleUrls: ['./sign-up.component.css'],
 })
 export class SignUpComponent {
-  constructor(private router: Router, private auth: AuthService) {}
+  fg: FormGroup;
+  constructor(
+    private fb: FormBuilder,
+    private auth: AuthService,
+    private router: Router
+  ) {
+    this.fg = fb.group({
+      name: ['', [Validators.required]],
+      email: ['', [Validators.required, Validators.email]],
+      password: [
+        '',
+        [Validators.required, Validators.minLength(6), passwordValidator()],
+      ],
+      cPassword: ['', [matchValidator('password', false)]],
```

Comment:
Password confirmation validation isn't showing any error message. Please test it and try to fix it. Also, you should validate that it is required.